### PR TITLE
Library upgrades, various fixes

### PR DESCRIPTION
**Pull request by adaamz**
- PHPStan upgrade to 0.12
- ninjify/qa upgrade to 0.10
- ninjify/nunjuck upgrade to 0.3
- PHPStan installation process/travis ci jobs/Makefile taken from [contributte/bare](https://github.com/contributte/bare/)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -42,13 +42,13 @@
     ],
     "phpstan-install": [
```
Comment (f3l1x) [inline composer.json]: Could you please synchronize it with contributte/bare repo? 

https://github.com/contributte/bare/blob/master/composer.json#L24-L27

Comment (adaamz) [reply composer.json]: Whole PHPStan installation process or just versions? :)

Comment (f3l1x) [reply composer.json]: Whole composer setup please.

Comment (adaamz) [reply composer.json]: ok, synced Makefile/.travis.yml/composer scripts in https://github.com/contributte/social/pull/6/commits/5d578de36a468c2bd598c0c6a3de9d1614db0fbd

**Comment by adaamz** [comment]
I noticed that in this repo, the docs directory is named `docs`, but in contribute/bare name is `.docs`. Also there in .gitattributes is export-ignored directory `.docs` (which does not exists).

Should I rename `docs` directory to `.docs`?

**Comment by f3l1x** [comment]
Awesome! Thanks.

**Comment by f3l1x** [comment]
I merged it. We can resolve docs in other PR. Yup, please rename it do `.docs` thank you.

**Pull request merged**